### DISCUSSION TITLE
Implement sprinting?

### DISCUSSION THREAD

**Discussion by GadyZ** [FreeFlyFall/RigidBodyController#3] [Ideas]
Hello! First of all, thank you so very much for making this. Physics based character controllers feel just perfect to me and I wish more people used them.
I am making this discussion to request a feature: sprinting.
I have tried my hand at this but I feel it is not ideal. Here's what I did.
`func is_below_speed_limit(nvel, vel):
	var current_sprinting_speed_limit
	match Input.is_action_pressed("sprint"):
		true:
			current_sprinting_speed_limit = sprinting_speed_limit
			#print(current_sprinting_speed_limit)
		false:
			current_sprinting_speed_limit = 0
			#print(current_sprinting_speed_limit)
	return ((nvel.x >= 0 and vel.x < nvel.x * (current_speed_limit + current_sprinting_speed_limit)) or (nvel.x <= 0 and vel.x > nvel.x * (current_speed_limit + current_sprinting_speed_limit)) or
		(nvel.z >= 0 and vel.z < nvel.z*(current_speed_limit + current_sprinting_speed_limit)) or (nvel.z <= 0 and vel.z > nvel.z*(current_speed_limit + current_sprinting_speed_limit)) or
		(nvel.x == 0 or nvel.z == 0))`

I added a sprinting speed to the maximum allowed speed, in your is_below_speed_limit to make use of the existing code, it's a pretty hacky solution, and the main issue with it is with slowing down. The player sprints and all, but does not come back to the normal walking speed fast enough. 
This I solved by reducing the friction decrease factor, so the speed does decrease faster, but it also decreases movement speed in general. 

Totally understandable if this is something you are not interested in working on.
Thank you for taking the time to read this :) EDIT: Just saw that the code block is all bugged.. here is a hastebin link for easier reading: https://hastebin.com/penezegope.yaml

**Comment by FreeFlyFall** [comment]
Thanks for the request. This one still has a way to go, but I hope it's useful.
I went ahead an added sprinting similar to how you described it in the latest commit.

I also noticed as you mentioned that the player wasn't slowing down quickly when moving because of the friction divider. The friction isn't being divided when grounded anymore after the last commit. It was there to prevent sticking before I switched to a new slope detection system, and isn't needed in that way anymore. The friction divider still works in the air and is mostly for keeping the player from sticking to walls when they push into them in the air and have a high amount of air control.
